modules/nw-ingress-controller-nodeportservice-projects.adoc

```diff
@@ -0,0 +1,195 @@
+----
+$ oc get svc -n openshift-ingress
+----
++
+.Example output that shows port `80:32432/TCP` for the `router-nodeport-custom-ic3` service
+[source,terminal]
+----
+NAME                        TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                     AGE
+router-internal-default     ClusterIP   172.30.195.74    <none>        80/TCP,443/TCP,1936/TCP                     223d
+router-nodeport-custom-ic3   NodePort    172.30.109.219   <none>        80:32432/TCP,443:31366/TCP,1936:30499/TCP   155m
+----
+
+. To create a new project, enter the following command:
++
+[source,terminal]
+----
+$ oc new-project <project_name>
+----
+
+. To label the new namespace, enter the following command:
++
+[source,terminal]
+----
+$ oc label namespace <project_name> <key>=<value> <1>
+----
+<1> Where `<key>=<value>` must match the value in the `namespaceSelector` section of your Ingress Controller CR.
+
+. Create a new application in your cluster:
++
+[source,terminal]
+----
+$ oc new-app --image=<image_name> <1>
+----
+<1> An example of `<image_name>` is `quay.io/openshifttest/hello-openshift:multiarch`.
+
+. Create a `Route` object for a service, so that the pod can use the service to expose the application external to the cluster.
++
+[source,terminal]
+----
+$ oc expose svc/<service_name> --hostname=<svc_name>-<project_name>.<custom_ic_domain_name> <1>
+----
++
+[NOTE]
+====
+You must specify the domain name of your custom Ingress Controller in the `--hostname` argument. If you do not do this, the Ingress Operator uses the default Ingress Controller to serve all the routes for your cluster. 
+====
+
+. Check that the route has the `Admitted` status and that it includes metadata for the custom Ingress Controller:
++
+[source,terminal]
+----
+$ oc get route/hello-openshift -o json | jq '.status.ingress
+----
++
+.Example output
+[source,terminal]
+----
+# ...
+{
+  "conditions": [
+    {
+      "lastTransitionTime": "2024-05-17T18:25:41Z",
+      "status": "True",
+      "type": "Admitted"
+    }
+  ],
+  [
+    {
+      "host": "hello-openshift.nodeportsvc.ipi-cluster.example.com",
+      "routerCanonicalHostname": "router-nodeportsvc.nodeportsvc.ipi-cluster.example.com",
+      "routerName": "nodeportsvc", "wildcardPolicy": "None"
+    }
+  ],
+}
+----
+
+. Update the default `IngressController` CR to prevent the default Ingress Controller from managing the `NodePort`-type `Service`. The default Ingress Controller will continue to monitor all other cluster traffic.
++
+[source,terminal]
+----
+$ oc patch --type=merge -n openshift-ingress-operator ingresscontroller/default --patch '{"spec":{"namespaceSelector":{"matchExpressions":[{"key":"<key>","operator":"NotIn","values":["<value>]}]}}}'
+----
+
+.Verification
+
+. Verify that the DNS entry can route inside and outside of your cluster by entering the following command:
++
+[source,terminal]
+----
+$ dig +short <svc_name>-<project_name>.<custom_ic_domain_name> <node_ip> <1>
```

Comment:
The <node_ip> is the result of `dig` command, it should appear on the next line.

Reply:
:heavy_check_mark: